src/vivarium_public_health/disease/special_disease.py

```diff
@@ -0,0 +1,134 @@
+import pandas as pd
+
+
+class RiskAttributableDisease:
+    """Component to model a disease fully attributed by a risk.
+    
+    For some (risk, cause) pairs with population attributable fraction
+    equal to 1, the clinical definition of the with condition state 
+    corresponds to a particular exposure of a risk.
+
+    For example, a diagnosis of ``diabetes_mellitus`` occurs after 
+    repeated measurements of fasting plasma glucose above 7 mmol/L. 
+    Similarly, ``protein_energy_malnutrition`` corresponds to a weight 
+    for height ratio that is more than two standard deviations below
+    the WHO guideline median weight for height.  In the Global Burden
+    of Disease, this corresponds to a categorical exposure to 
+    ``child_wasting`` in either ``cat1`` or ``cat2``.
+    
+    The definition of the disease in terms of exposure should be provided
+    in the ``threshold`` configuration flag.  For risks with continuous
+    exposure models, the threshold should be provided as a single 
+    ``float`` or ``int``.  For categorical risks, the threshold
+    should be provided as a list of categories.
+    
+    In addition to the threshold level, you may configure whether
+    there is any mortality associated with this disease with the 
+    ``mortality`` configuration flag.
+     
+    Finally, you may specify whether the someone should "recover" 
+    from the disease if their exposure level falls outside the 
+    provided threshold. 
+    
+    In our provided examples, a person would no longer be experiencing 
+    ``protein_energy_malnutrition`` if their exposure drift out (or
+    changes via an intervention) of the provided exposure categories. 
+    Having your ``fasting_plasma_glucose`` drop below a provided level
+    does not necessarily mean you're no longer diabetic however.
+
+    Configuration defaults should be given as, for the continuous risk factor,
+
+    diabetes_mellitus:
+        threshold : 7
+        mortality : True
+        recoverable : False
+
+    For the categorical risk factor,
+
+    protein_energy_malnutrition:
+        threshold : ['cat1', 'cat2'] # provide the categories to get PEM.
+        mortality : True
+        recoverable : True
+    """
+
+    configuration_defaults = {
+        'risk_attributable_disease': {
+            'threshold': None,
+            'mortality': True,
+            'recoverable': True
+        }
+    }
+
+    def __init__(self, name, risk):
```

Comment:
For ordering functions in a component, I always try to follow the pattern:

1. __init__
2. setup
3. on_initialize_simulants
4. time_step_handler
5. on_collect_metrics_handler
6. helper functions
7. metrics pipeline modifier